app/src/main/java/io/github/fate_grand_automata/ui/launcher/BattleLauncher.kt

```diff
@@ -59,16 +65,79 @@ fun battleLauncher(
             }
     }
     var selectedConfigIndex by remember { mutableStateOf(configs.indexOf(prefs.selectedBattleConfig)) }
-    var refillResources by remember { mutableStateOf(prefs.refill.resources.toSet()) }
-    var refillCount by remember { mutableStateOf(prefs.refill.repetitions) }
-    var shouldLimitRuns by remember { mutableStateOf(prefs.refill.shouldLimitRuns) }
-    var limitRuns by remember { mutableStateOf(prefs.refill.limitRuns) }
-    var shouldLimitMats by remember { mutableStateOf(prefs.refill.shouldLimitMats) }
-    var limitMats by remember { mutableStateOf(prefs.refill.limitMats) }
-    var shouldLimitCEs by remember { mutableStateOf(prefs.refill.shouldLimitCEs) }
-    var limitCEs by remember { mutableStateOf(prefs.refill.limitCEs) }
+
+    val perServerConfigPref by remember {
+        mutableStateOf(
+            prefs.perServerConfigPrefList.single { selected ->
+                prefs.gameServer == selected.server
```

Comment:
A NoSuchElementException is thrown here if you use game servers for BetterFGO.

```suggestion
                prefs.gameServer.simple == selected.server.simple
```